src/resolution.rs

```diff
@@ -4,11 +4,10 @@ use std::{
     sync::Arc,
 };
 
-use crate::package_json::PackageJson;
+use crate::{Cache, PackageJson};
 
 /// The final path resolution with optional `?query` and `#fragment`
-#[derive(Clone)]
-pub struct Resolution {
+pub struct Resolution<C: Cache> {
```

Comment:
We may need to provide a type alias for `Resolution<FsCache<FileSystemOs>>` to reduce some API changes down stream

```
error[E0107]: missing generics for struct `Resolution`
   --> crates/rolldown_plugin_vite_resolve/src/resolver.rs:235:29
    |
235 |   ) -> Result<oxc_resolver::Resolution, oxc_resolver::ResolveError> {
    |                             ^^^^^^^^^^ expected 1 generic argument
```

Reply:
Added `FsResolution` type alias.